src/setup.vala

```diff
@@ -564,152 +569,255 @@ class SetupDialog : Gtk.Dialog {
-            var index = variant.get_int32 ();
-            int _index;
             do {
-                model.get (iter, column, out _index, -1);
-                if (index == _index) {
-                    combo.set_active_iter (iter);
+                string _str;
+                model.get (iter, column, out _str, -1);
+                if (str == _str) {
+                    value.set_int(index);
+                    found = true;
                     break;
                 }
+                index++;
             } while (model.iter_next (ref iter));
         }
 
-        combo.changed.connect (() => {
-                save_combobox (name, combo, column);
-            });
+        if (!found)
+            warning ("Not found: %s".printf(str));
+        return found;
     }
 
-    void load_combobox_string (string name,
-                               Gtk.ComboBox combo,
-                               int column) {
-        Variant? variant = preferences.get (name);
-        assert (variant != null);
+    static Variant combobox_string_set_mapping (Value value,
+                                                VariantType expected_type,
+                                                SettingsBindData data) {
+        Gtk.ComboBox combo = data.combo;
+        int column = data.column;
         Gtk.TreeIter iter;
+        if (combo.get_active_iter (out iter)) {
+            string str;
+            var model = combo.get_model ();
+            model.get (iter, column, out str, -1);
+            return str;
+        }
+        assert_not_reached ();
+    }
+
+    static bool combobox_enum_get_mapping (Value value,
+                                           Variant variant,
+                                           SettingsBindData data) {
+        bool found = false;
+        Gtk.ComboBox combo = data.combo;
+        int column = data.column;
+        unowned EnumClass eclass = data.eclass;
+        string nick = variant.get_string(null);
+        var evalue = eclass.get_value_by_nick (nick);
+        int mode = evalue.value;
+        int index = 0;
         var model = combo.get_model ();
+        Gtk.TreeIter iter;
         if (model.get_iter_first (out iter)) {
-            string str = variant.get_string ();
             do {
-                string _str;
-                model.get (iter, column, out _str, -1);
-                if (str == _str) {
-                    combo.set_active_iter (iter);
+                int _mode;
+                model.get (iter, column, out _mode, -1);
+                if (mode == _mode) {
+                    value.set_int(index);
+                    found = true;
                     break;
                 }
+                index++;
             } while (model.iter_next (ref iter));
         }
 
-        combo.changed.connect (() => {
-                save_combobox_string (name, combo, column);
-            });
+        if (!found)
+            warning ("Not found: %i".printf(mode));
```

Comment:
Change this to `assert_if_reached`. The user can't use undefined enum value in gsettings.